### DISCUSSION TITLE
How can I publish DataportalServer project automatically?

### DISCUSSION THREAD

**Discussion by amselem** [MarimerLLC/csla#3249] [General]
Hello

We've a Winforms app deployed via ClickOnce.  Some businessobjects are configured to go through a remote dataportal hosted in an IIS server.  Both applications are manually deployed to the same server using "publish" command from Visual Studio.

Our problem is that we forget sometimes to update the dataportal App  (most of the updates go to the UI or to local BusinessObjects , so the application won't fail if the dataportal isn't updated).  
But when there's a code change to a remote Business Object and the Dataportal is not updated the application fails as it doesn't have the same library version in client and server.

Do you know how can I automatically publish the Dataportal web app when publishing the ClickOnce app?
Another option would be to periodically execute an script to update the dataportal copying the DLLs from the published clickonce application.   

Any suggestions would be appreciated.

**Comment by Chicagoan2016** [comment]
Brian Noyes had written an excellent book about ClickOnce years (decades lol) ago. I have a copy laying around somewhere in my office. I am working from home today but hopefully I could swing by later; weather permitting.
